### ISSUE TITLE
Parameters not supported for SemanticVersion in AWS::Serverless::Application resource

### ISSUE THREAD

**Issue by mixja** [awslabs/serverless-application-model#897]
**Description:**

When you attempt to reference a parameter for the `SemanticVersion` property of the `Location` object, SAM processing fails with an internal transform failure.

**Steps to reproduce the issue:**
1.  Attempt to create a change set/stack for the following SAM template:

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Description: Some Template
Parameters:
  SecretsProvisionerApplicationVersion:
    Description: Secrets Provisioner Version
    Type: String
Resources:
  SecretsProvisionerApplication:
    Type: AWS::Serverless::Application
    Properties:
      Location:
        ApplicationId: arn:aws:serverlessrepo:ap-southeast-2:012345678901:applications/secrets-provisioner
        SemanticVersion: !Ref SecretsProvisionerApplicationVersion
```

2.  Provide valid version for semantic version parameter

**Observed result:**

Change set/stack creation fails with error `Transform AWS::Serverless-2016-10-31 failed with: Internal transform failure.`

**Expected result:**

You should be able to parameterise the target semantic version of a Serverless application resource.

If you hardcode the `SemanticVersion` in the resource as per below, change set/stack creation is successful:

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Description: Some Template
Resources:
  SecretsProvisionerApplication:
    Type: AWS::Serverless::Application
    Properties:
      Location:
        ApplicationId: arn:aws:serverlessrepo:ap-southeast-2:012345678901:applications/secrets-provisioner
        SemanticVersion: 0.1.17
```

**Comment by keetonian** [comment]
Try it again :) this should have been fixed in [SAM v1.11.0](https://github.com/awslabs/serverless-application-model/releases/tag/v1.11.0) which was just released. 

I just verified that this works just in case with the following template:
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Parameters:
  Semver:
    Type: String
    Default: 1.0.3
Resources:
  helloworld:
    Type: AWS::Serverless::Application
    Properties:
      Location:
        ApplicationId: arn:aws:serverlessrepo:us-east-1:077246666028:applications/hello-world
        SemanticVersion: !Ref Semver
      Parameters: 
        IdentityNameParameter: YOUR_VALUE  
```

**Comment by mixja** [comment]
Thanks - looks like this is fixed from CloudFormation side of things, but SAM CLI is not working as yet (imagine this will catch up soon).
